src/joint_state_listener.cpp

```diff
@@ -114,22 +114,22 @@ void JointStateListener::callbackJointState(const JointStateConstPtr& state)
   }
 
   // check if we moved backwards in time (e.g. when playing a bag file)
-  ros::Time now = ros::Time::now();
-  if (last_callback_time_ > now) {
+  ros::Time time_now = ros::Time::now();
+  if (last_callback_time_ > time_now) {
     // force re-publish of joint transforms
     ROS_WARN("Moved backwards in time (probably because ROS clock was reset), re-publishing joint transforms!");
     last_publish_time_.clear();
   }
   ros::Duration warning_threshold(30.0);
-  if ((state->header.stamp + warning_threshold) < now) {
-    ROS_WARN_THROTTLE(10, "Received JointState is %f seconds old.", (now - state->header.stamp).toSec());
+  if ((state->header.stamp + warning_threshold) < time_now) {
+    ROS_WARN_THROTTLE(10, "Received JointState is %f seconds old.", (time_now - state->header.stamp).toSec());
   }
-  last_callback_time_ = now;
+  last_callback_time_ = time_now;
 
   // determine least recently published joint
-  ros::Time last_published = now;
-  for (size_t i = 0; i < state->name.size(); ++i) {
-    ros::Time t = last_publish_time_[state->name[i]];
+  ros::Time last_published = time_now;
+  for (const auto & i : state->name) {
```

Comment:
Let's not use `auto`; it's hard to tell what the type is supposed to be then.  I guess this would be `const std::string & name` then.